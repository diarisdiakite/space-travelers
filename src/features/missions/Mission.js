import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams, Link } from 'react-router-dom';
import { selectMissionsById, joinMission, cancelMissionParticipation } from './missionsSlice';

function Mission() {
  const dispatch = useDispatch();
  const { missionId } = useParams();
  const mission = useSelector((state) => selectMissionsById(state, missionId));

  return (
    <div className="main-container">
      <div className="mission-card">
        <div className="mission" key={`mission-${missionId}`}>
          <h3>{mission?.mission_name}</h3>
          <p>{mission?.description}</p>
          <p><Link to={mission?.website}>{mission?.website}</Link></p>
          <button
            type="button"
            aria-label="Update mission"
            onClick={() => dispatch(joinMission())}
          >
            Reserve Mission
          </button>
          <button
            type="button"
            aria-label="Remove mission"
            onClick={() => dispatch(cancelMissionParticipation())}
          >
            Cancel Reservation
          </button>
        </div>
      </div>
    </div>
  );
}

export default Mission;
