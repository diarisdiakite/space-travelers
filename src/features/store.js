import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './missions/missionsSlice';
import usersReducer from './users/usersSlice';
import rocketsSlice from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    users: usersReducer,
    rockets: rocketsSlice,
  },
});

export default store;
