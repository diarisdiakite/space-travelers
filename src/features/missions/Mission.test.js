import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import store from '../store';
import Mission from './Mission';

describe('Mission component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <Mission />
          </MemoryRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
