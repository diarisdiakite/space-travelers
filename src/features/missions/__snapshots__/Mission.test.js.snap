// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission component renders correctly 1`] = `
<div
  className="main-container"
>
  <div
    className="mission-card"
  >
    <div
      className="mission"
    >
      <h3 />
      <p />
      <p>
        <a
          href="/"
          onClick={[Function]}
        />
      </p>
      <button
        aria-label="Update mission"
        onClick={[Function]}
        type="button"
      >
        Reserve Mission
      </button>
      <button
        aria-label="Remove mission"
        onClick={[Function]}
        type="button"
      >
        Cancel Reservation
      </button>
    </div>
  </div>
</div>
`;
